### PR TITLE
(PC-6703) insuffisant credit duo section

### PR DESCRIPTION
**Pull request by emmajmn**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-6703

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [x] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] |
| -------------------- | :----------------------: |
| image | image |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/bookOffer/components/__tests__/BookHourChoice.test.tsx [hunk 1]
```diff
@@ -110,7 +110,7 @@ describe('BookHourChoice', () => {
     expect(secondPrice.props.children).toBe('épuisé')
   })
 
-  fit("should show 'crédit insuffisant' if not enough credit", () => {
+  it("should show 'crédit insuffisant' if not enough credit", () => {
```
Comment (antoinewg) [inline src/features/bookOffer/components/__tests__/BookHourChoice.test.tsx:113]: 👍

File: src/features/bookOffer/components/__tests__/BookDuoChoice.test.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import { act, fireEvent, render } from '@testing-library/react-native'
+import React from 'react'
+
+import { mockOffer } from 'features/bookOffer/fixtures/offer'
+import { BookingState, Step } from 'features/bookOffer/pages/reducer'
+import { reactQueryProviderHOC } from 'tests/reactQueryProviderHOC'
+
+import { BookDuoChoice } from '../BookDuoChoice'
+
+allowConsole({ log: true })
```
Comment (antoinewg) [inline src/features/bookOffer/components/__tests__/BookDuoChoice.test.tsx:10]: tu peux l'enlever ? Le but est de se débarasser des console dans les tests. Donc si on rajoute des tests avec des logs, on a pas fini.

File: src/features/bookOffer/components/__tests__/BookDuoChoice.test.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      date: new Date('2021-03-02T20:00:00'),
+    } as BookingState,
+    dismissModal: mockDismissModal,
+  })),
+  useBookingStock: jest.fn(() => ({
+    price: 2000,
+    id: '148409',
+    beginningDatetime: new Date('2021-03-02T20:00:00'),
+  })),
+  useBookingOffer: jest.fn(() => mockOffer),
```
Comment (antoinewg) [inline src/features/bookOffer/components/__tests__/BookDuoChoice.test.tsx:31]: t'utilise pas ça si ?

File: src/features/bookOffer/components/__tests__/BookDuoChoice.test.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    const page = render(reactQueryProviderHOC(<BookDuoChoice />))
+
+    const soloChoice = page.queryByTestId('DuoChoice1')
+
+    if (soloChoice) {
+      act(() => fireEvent.press(soloChoice))
+
+      expect(mockDispatch).toHaveBeenCalledWith({ type: 'SELECT_QUANTITY', payload: 1 })
+    } else {
+      throw new Error('should have find solo choice')
```
Comment (antoinewg) [inline src/features/bookOffer/components/__tests__/BookDuoChoice.test.tsx:63]: il devrait pas y avoir de if dans un test a priori, à part si c'est pour faire diverger ce que tu tests. Si le test fail faut investiguer pk.

File: src/features/bookOffer/components/__tests__/BookDuoChoice.test.tsx (hunk 2 above)
Comment (emmajmn) [reply src/features/bookOffer/components/__tests__/BookDuoChoice.test.tsx:10]: bien vu !

**Review by antoinewg — APPROVED** [review]
voilà 👍

**Pull request merged**